sdks/python/apache_beam/io/fileio.py

```diff
@@ -270,6 +295,8 @@ def _rename_batch(batch):
       """_rename_batch executes batch rename operations."""
       source_files, destination_files = batch
       exceptions = []
+      if self._file_system is None:
```

Comment:
I agree that that should be the actual check

```
if self._file_system is None:
  self._file_system = get_filesystem(file_path_prefix)
```
However, I think I didn't need to add it in the first place, since the check has already happened in open (and this is at finalize_write).